neo/SmartContract/Manifest/ContractManifest.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Neo.IO;
+using Neo.IO.Json;
+using System.IO;
+using System.Linq;
+
+namespace Neo.SmartContract.Manifest
+{
+    /// <summary>
+    /// When a smart contract is deployed, it must explicitly declare the features and permissions it will use.
+    /// When it is running, it will be limited by its declared list of features and permissions, and cannot make any behavior beyond the scope of the list.
+    /// </summary>
+    public class ContractManifest : ISerializable
+    {
+        /// <summary>
+        /// Max length for a valid Contract Manifest
+        /// </summary>
+        public const int MaxLength = ushort.MaxValue;
```

Comment:
It's too large for a manifest, I think. Maybe `2048` is enough.

Reply:
maybe it's enough, because if you want to whitelist some DEX, they can create a Group together